sch_yang_condition.c

```diff
@@ -0,0 +1,923 @@
+                                int depth, bool or_self, int *flags)
+{
+    xmlChar *idref_prefix;
+    xmlChar *idref_self;
+    xmlChar *name;
+    GList *iter;
+    xpath_node *node;
+    sch_node *s_node = NULL;
+    yang_result result = { };
+
+    /* The list should have 2 members */
+    presult->result = false;
+
+    if (g_list_length (xnode->arg_list) != 2)
+        return;
+
+    iter = g_list_first (xnode->arg_list);
+    if (iter)
+    {
+        node = (xpath_node *) iter->data;
+        sch_yang_process_xnode (instance, root, path, step_path, node, &result,
+                                depth + 1, flags);
+        if (result.result)
+            s_node = sch_lookup (instance, result.value);
+
+        g_free (result.value);
+        result.value = NULL;
+        iter = iter->next;
+        if (result.result && iter)
+        {
+            node = (xpath_node *) iter->data;
+            sch_yang_process_xnode (instance, root, path, step_path, node, &result,
+                                    depth + 1, flags);
+            if (!result.result)
+            {
+                g_free (result.value);
+                return;
+            }
+        }
+    }
+    else
+    {
+        return;
+    }
+
+    if (!s_node)
+    {
+        g_free (result.value);
+        return;
+    }
+
+    name = xmlGetProp ((xmlNode *) s_node, BAD_CAST "name");
+    idref_prefix = xmlGetProp ((xmlNode *) s_node, BAD_CAST "idref_prefix");
+    if (idref_prefix)
+    {
+        char *test_value;
+
+        test_value = g_strdup_printf ("%s:%s", (char *) idref_prefix, name);
+        if (g_strcmp0 (test_value, result.value) == 0)
+            presult->result = true;
+
+        g_free (test_value);
+        xmlFree (idref_prefix);
+    }
+    else if (or_self)
+    {
+        idref_self = xmlGetProp ((xmlNode *) s_node, BAD_CAST "idref_self");
+        if (idref_self)
+        {
+            char *test_value;
+            name = xmlGetProp ((xmlNode *) s_node, BAD_CAST "name");
+
+            test_value = g_strdup_printf ("%s:%s", (char *) idref_self, name);
+            if (g_strcmp0 (test_value, result.value) == 0)
+            {
+                presult->result = true;
+            }
+
+            g_free (test_value);
+            xmlFree (idref_prefix);
+        }
+    }
+    xmlFree (name);
+    g_free (result.value);
+}
+
+static void
+sch_yang_process_arg_list (sch_instance *instance, GNode *root, char *path,
+                           char *step_path, xpath_node *xnode, yang_result *presult,
+                           int depth, int *flags)
+{
+    GList *iter;
+    xpath_node *lnode;
+    yang_result result = { };
+    bool first = true;
+
+    presult->result = false;
+    for (iter = g_list_first (xnode->arg_list); iter; iter = g_list_next (iter))
+    {
+        lnode = (xpath_node *) iter->data;
+        sch_yang_process_xnode (instance, root, path, step_path, lnode, &result,
```

Comment:
Does result.result get cleared inside sch_yang_process_xnode?

Reply:
You answered your own question.